### PR TITLE
Bugs 1183885 and 1191979 regarding offset and limit.

### PR DESCRIPTION
**Pull request by pelmers**
The first commit of this pair was already reviewed in https://github.com/mozilla/dxr/pull/455, but it was dropped from the master branch most likely when I accidentally temporarily merged to path promotion.

To fix 1191979 (the second commit here), I string.replace offset= and limit= by '' from the ajax URL before updating the history state.

### CODE REVIEW

File: dxr/static/js/dxr.js [hunk 1]
```diff
@@ -426,23 +391,30 @@ $(function() {
...
                 data.query = query;
-                // New results, overwrite
+                // New results, display them.
                 if (myRequestNumber > displayedRequestNumber) {
                     displayedRequestNumber = myRequestNumber;
-                    populateResults(data, false);
-                    var pushHistory = function() {
-                        history.pushState({}, '', queryString);
+                    populateResults(data, appendResults);
+                    var pushHistory = function () {
```
Comment (erikrose) [inline dxr/static/js/dxr.js]: Can you think of any reason we use this cumbersome function definition syntax rather than just `function pushHistory`?

Comment (pelmers) [inline dxr/static/js/dxr.js]: They behave the same in this case, so I kept what we had originally. I shall change to the other syntax.

File: dxr/static/js/dxr.js [hunk 2]
```diff
@@ -426,23 +391,30 @@ $(function() {
...
+                        // Strip off offset= and limit= when updating.
+                        var displayURL = queryString.replace(/&?offset=\d+/, '').replace(/&?limit=\d+/, '');
```
Comment (erikrose) [inline dxr/static/js/dxr.js]: The regex should insist on a ? or a & before each param. Otherwise, we'll run into trouble when we inevitably add a foo_offset param later. (Better still might be always building up a query URL from scratch in doQuery, but this is good enough for a quick fix of a user-visible problem.)

File: dxr/static/js/dxr.js (hunk 1 above)
Comment (erikrose) [inline dxr/static/js/dxr.js]: I didn't mean you had to change it as part of this PR; I just wondered if you saw anything I didn't. It's better not to change it (or at least to keep it as a separate commit).

**Comment by erikrose** [comment]
Looking good. :-)
